### DISCUSSION TITLE
Clear cache for a single room and/or clear media based on retention on the server

### DISCUSSION THREAD

**Discussion by weilhr** [element-hq/element-meta#1448] [Ideas]
At the moment Element on iOS only supports to clear the cache globaly. I have a room getting notifications which are including pictures (security camera). In synapse the messages in the room are set to retire. Though the message itself is not available after retired, the photos are still in cache and would need to be deleted individually. It would be great if elements allow to either
- clear the cache for the particular room (not for all rooms globally)
- enable automatic clearing media data based on the set rentention on the server.

**Comment by t3chguy** [comment]
> clear the cache for the particular room (not for all rooms globally)

Not feasible as the Room specific `/sync` API has been deprecated

> enable automatic clearing media data based on the set rentention on the server.

https://github.com/vector-im/element-meta/discussions/730

**Reply by weilhr** [reply]
@t3chguy, thanks for you fast response and quick reply. Indeed, #730 is close to what I was proposing. At least for me the server based retention works for the message, but not for media sent into the room. When #730 also covers retention of attached media, we can close this ER as dup.
